openeo-geotrellis/src/main/scala/org/openeo/geotrellis/geotiff/package.scala

```diff
@@ -838,31 +836,47 @@ package object geotiff {
       uploadToS3(tempFile, correctS3Path)
 
     } else {
-      val tempFile = Files.createTempFile(null, ".tif")
+      val tempFile = getTempFile(null, ".tif")
       // TODO: Try to run fsync on the file opened by GeoTrellis (without the temporary copy)
       geoTiff.write(tempFile.toString, optimizedOrder = true)
 
-      // Geotrellis writes the file piecewise and sometimes files are only partially written.
-      // Maybe a move operation is easier for the fusemount:
-      try {
-        Files.move(tempFile, Path.of(path))
-      } catch {
-        case e: FileAlreadyExistsException =>
-          logger.info("FileAlreadyExistsException. Will overwrite file: " + e.getMessage)
-          // The existing file could be a partial result of a previous failing Spark task.
-          Files.deleteIfExists(Path.of(path))
-          Files.move(tempFile, Path.of(path))
-      }
+      moveOverwriteWithRetries(tempFile, Path.of(path))
 
       // Call fsync on the parent path to assure the fusemount is up-to-date.
       // The equivalent of Python's os.fsync
-      FileChannel.open(Path.of(path)).force(true)
-
+      try {
+        FileChannel.open(Path.of(path)).force(true)
+      } catch {
+        case _: NoSuchFileException => // Ignore. The file may already be deleted by another executor
+      }
       path
     }
 
   }
 
+  def moveOverwriteWithRetries(oldPath: Path, newPath: Path): Unit = {
+    var try_count = 1
+    breakable {
+      while (true) {
+        try {
+          if (newPath.toFile.exists()) {
+            // It might be a partial result of a previous failing task.
```

Comment:
Isn't it safer to just assume that the file is correctly written, or being written when the path already exists?

Reply:
I did not encounter it, but feel that the file might be a partial written result before that executor went OOM. 
OOM is a frequent reason why a second executor is started